lib/types.ts

```diff
@@ -158,7 +164,7 @@ export class TUnion extends TType {
   private _failMsg: string;
   constructor(public ttypes: TType[]) {
     super();
-    const names = ttypes.map((t) => t instanceof TName || t instanceof TLiteral ? t.name : null)
+    const names = ttypes.map((t) => t instanceof TName || t instanceof TLiteral || t instanceof TArray ? t.name : null)
       .filter((n) => n);
     const otherTypes: number = ttypes.length - names.length;
     if (names.length) {
```

Comment:
It's not required for this PR but I think adding a name for unions looks pretty straightforward when `otherTypes === 0`.